test/spectral/tests/spectral-validator.test.js

```diff
@@ -0,0 +1,102 @@
+const { getCapturedText } = require('../../test-utils');
+const spectralValidator = require('../../../src/spectral/utils/spectral-validator');
+
+describe('spectral - test spectral-validator.js', function() {
+  let consoleSpy;
+
+  beforeEach(() => {
+    consoleSpy = jest.spyOn(console, 'log').mockImplementation(() => {});
+  });
+
+  afterEach(() => {
+    consoleSpy.mockRestore();
+  });
+
+  it('test should parse mock spectral result as an error', function() {
+    // Create a mock Spectral result object
+    const mockResult = [
+      {
+        code: 'operation-tags',
+        message: 'Operation should have non-empty `tags` array.',
+        path: ['paths', '/pet', 'put'],
+        severity: 0,
+        start: { line: 0, character: 0 },
+        end: { line: 134, character: 15 }
+      }
+    ];
+    const debug = false;
+    const parsedSpectralResults = spectralValidator.parseResults(
+      mockResult,
+      debug
+    );
+    const capturedText = getCapturedText(consoleSpy.mock.calls);
+    const allOutput = capturedText.join('');
+
+    expect(parsedSpectralResults.errors.length).toEqual(1);
+    expect(parsedSpectralResults.warnings.length).toEqual(0);
```

Comment:
I don't think you need this for every test but for at least one test, it would be good to make expectations about the message and path to verify they are parsed correctly